src/main/java/com/appdirect/sdk/credentials/SourceBackedCredentials.java

```diff
@@ -0,0 +1,24 @@
+package com.appdirect.sdk.credentials;
+
+import com.appdirect.sdk.appmarket.Credentials;
+import com.appdirect.sdk.appmarket.DeveloperSpecificAppmarketCredentialsSupplier;
+
+import java.util.function.Function;
+
+import static com.appdirect.sdk.appmarket.Credentials.invalidCredentials;
+
+/**
+ * Functionality to find Credentials by key from a different sources: Database, ResourceFile, in memory cache, etc.
+ */
+public class SourceBackedCredentials implements DeveloperSpecificAppmarketCredentialsSupplier {
```

Comment:
Something I didn't see before, you should follow the interface convention for the name of the class:
```suggestion
public class SourceBackedCredentialsSupplier implements DeveloperSpecificAppmarketCredentialsSupplier {
```